File: frontend/src/lib/crm_pipeline.ts
import axios from "axios";

const API = (process.env.NEXT_PUBLIC_API_URL ?? "http://localhost:8000") + "/api/v1";
const r = <T>(p: Promise<{ data: T }>) => p.then((res) => res.data);
const ax = axios.create({ baseURL: API });

// ── Types ─────────────────────────────────────────────────────────────────────

export type CRMRecordType   = "LEAD" | "OPPORTUNITY";
export type CRMAccountType  = "PROSPECT" | "CUSTOMER" | "DISTRIBUTOR" | "RETAILER" | "MODERN_TRADE" | "INSTITUTIONAL" | "SUPPLIER_XSELL" | "OTHER";
export type CRMSourceType   = "WEBSITE" | "REFERRAL" | "FIELD_SALES" | "EVENT" | "OUTBOUND" | "INBOUND" | "CAMPAIGN" | "MANUAL" | "IMPORT";
export type CRMStageType    = "LEAD" | "OPPORTUNITY" | "TERMINAL" | "NURTURING";
export type CRMTemperature  = "COLD" | "WARM" | "HOT";
export type CRMStatus       = "OPEN" | "WON" | "LOST" | "ON_HOLD" | "ARCHIVED";
export type CRMActivityType = "CALL" | "EMAIL" | "MEETING" | "WHATSAPP" | "VISIT" | "TASK" | "NOTE" | "DEMO" | "SAMPLE_SENT" | "QUOTATION_SENT" | "NEGOTIATION" | "COMPLAINT_FOLLOWUP" | "OTHER";
export type CRMActivityResult = "PLANNED" | "COMPLETED" | "NO_RESPONSE" | "RESCHEDULED" | "CANCELLED";
export type CRMLossReason   = "PRICE_TOO_HIGH" | "COMPETITOR_RELATION" | "PRODUCT_MISMATCH" | "TIMELINE_ISSUE" | "NO_BUDGET" | "NO_RESPONSE" | "COMPETITOR_PROMO" | "DISTRIBUTOR_CONFLICT" | "CAPACITY_SUPPLY" | "INTERNAL_FOLLOWUP" | "OTHER";
export type CRMWinReason    = "BETTER_PRICE" | "BETTER_PRODUCT_FIT" | "FASTER_RESPONSE" | "STRONGER_RELATION" | "AVAILABILITY" | "TECHNICAL_SUPPORT" | "PROMOTION_SUPPORT" | "OTHER";
export type CRMAIAgentType  = "LEAD_PRIORITIZATION" | "PIPELINE_RISK" | "WIN_LOSS_INSIGHT";
export type CRMAIRecStatus  = "PENDING" | "ACKNOWLEDGED" | "ACTIONED" | "DISMISSED";

export interface CRMStage {
  id: string;
  stage_code: string;
  stage_name: string;
  stage_type: CRMStageType;
  default_probability: number;
  sequence_no: number;
  active_flag: boolean;
  color_code?: string;
  notes?: string;
  created_at?: string;
}

export interface CRMInterestLine {
  id: string;
  crm_record_id: string;
  item_id?: string;
  item_name?: string;
  category_id?: string;
  brand_id?: string;
  expected_qty?: number;
  expected_frequency?: string;
  expected_value?: number;
  competitive_brand?: string;
  competitor_price?: number;
  notes?: string;
}

export interface CRMActivity {
  id: string;
  crm_record_id: string;
  activity_type: CRMActivityType;
  subject: string;
  description?: string;
  activity_datetime?: string;
  due_datetime?: string;
  completed_datetime?: string;
  assigned_to?: string;
  created_by?: string;
  result_status: CRMActivityResult;
  outcome_code?: string;
  next_step?: string;
  notes?: string;
  created_at?: string;
}

export interface CRMCompetitor {
  id: string;
  crm_record_id: string;
  competitor_name: string;
  competitor_product?: string;
  offer_type?: string;
  competitor_price?: number;
  competitor_strength?: string;
  competitor_weakness?: string;
  risk_level?: string;
  notes?: string;
  created_at?: string;
}

export interface CRMWinLoss {
  id: string;
  crm_record_id: string;
  final_status: string;
  close_date?: string;
  final_revenue?: number;
  final_margin?: number;
  loss_reason_code?: CRMLossReason;
  win_reason_code?: CRMWinReason;
  competitor_name?: string;
  narrative?: string;
  recorded_by?: string;
  notes?: string;
}

export interface CRMRecord {
  id: string;
  record_type: CRMRecordType;
  lead_code?: string;
  opportunity_code?: string;
  account_type: CRMAccountType;
  company_name: string;
  contact_person_name?: string;
  contact_email?: string;
  contact_phone?: string;
  customer_id?: string;
  distributor_id?: string;
  source_type: CRMSourceType;
  region_id?: string;
  channel_id?: string;
  assigned_rep_id?: string;
  assigned_team_id?: string;
  stage_id?: string;
  stage?: CRMStage;
  probability_pct?: number;
  expected_close_date?: string;
  expected_revenue?: number;
  expected_margin?: number;
  currency: string;
  lead_score: number;
  temperature: CRMTemperature;
  next_action_date?: string;
  status: CRMStatus;
  qualified_flag: boolean;
  qualified_date?: string;
  converted_date?: string;
  linked_quotation_id?: string;
  linked_order_id?: string;
  created_by?: string;
  notes?: string;
  created_at?: string;
  updated_at?: string;
  interest_lines?: CRMInterestLine[];
  activities?: CRMActivity[];
  competitors?: CRMCompetitor[];
  win_loss?: CRMWinLoss;
}

export interface CRMAIRec {
  id: string;
  agent_type: CRMAIAgentType;
  crm_record_id?: string;
  title: string;
  body: string;
  priority?: string;
  action_label?: string;
  status: CRMAIRecStatus;
  acknowledged_by?: string;
  notes?: string;
  created_at?: string;
}

export interface CRMDashboard {
  total_leads: number;
  total_opportunities: number;
  open_records: number;
  won_this_month: number;
  lost_this_month: number;
  total_pipeline_value: number;
  weighted_pipeline_value: number;
  overdue_activities: number;
  hot_records: number;
  conversion_rate_pct: number;
  avg_deal_size: number;
  stage_distribution: { stage: string; color: string; count: number; value: number }[];
  top_reps: { rep: string; count: number; value: number }[];
}

// ── Label Maps ─────────────────────────────────────────────────────────────────

export const STAGE_TYPE_LABELS: Record<CRMStageType, string> = {
  LEAD: "Lead Stage",
  OPPORTUNITY: "Opportunity Stage",
  TERMINAL: "Terminal",
  NURTURING: "Nurturing",
};

export const TEMPERATURE_LABELS: Record<CRMTemperature, string> = {
  COLD: "Cold",
  WARM: "Warm",
  HOT: "Hot",
};

export const TEMPERATURE_COLORS: Record<CRMTemperature, string> = {
  COLD: "#6B7280",
  WARM: "#F59E0B",
  HOT: "#EF4444",
};

export const STATUS_LABELS: Record<CRMStatus, string> = {
  OPEN: "Open",
  WON: "Won",
  LOST: "Lost",
  ON_HOLD: "On Hold",
  ARCHIVED: "Archived",
};

export const STATUS_COLORS: Record<CRMStatus, string> = {
  OPEN: "#3B82F6",
  WON: "#10B981",
  LOST: "#EF4444",
  ON_HOLD: "#F59E0B",
  ARCHIVED: "#6B7280",
};

export const ACTIVITY_TYPE_LABELS: Record<CRMActivityType, string> = {
  CALL: "Call",
  EMAIL: "Email",
  MEETING: "Meeting",
  WHATSAPP: "WhatsApp",
  VISIT: "Site Visit",
  TASK: "Task",
  NOTE: "Note",
  DEMO: "Demo",
  SAMPLE_SENT: "Sample Sent",
  QUOTATION_SENT: "Quotation Sent",
  NEGOTIATION: "Negotiation",
  COMPLAINT_FOLLOWUP: "Complaint Follow-up",
  OTHER: "Other",
};

export const ACTIVITY_RESULT_LABELS: Record<CRMActivityResult, string> = {
  PLANNED: "Planned",
  COMPLETED: "Completed",
  NO_RESPONSE: "No Response",
  RESCHEDULED: "Rescheduled",
  CANCELLED: "Cancelled",
};

export const SOURCE_LABELS: Record<CRMSourceType, string> = {
  WEBSITE: "Website",
  REFERRAL: "Referral",
  FIELD_SALES: "Field Sales",
  EVENT: "Event / Trade Show",
  OUTBOUND: "Outbound Call",
  INBOUND: "Inbound Inquiry",
  CAMPAIGN: "Campaign",
  MANUAL: "Manual Entry",
  IMPORT: "Import",
};

export const LOSS_REASON_LABELS: Record<CRMLossReason, string> = {
  PRICE_TOO_HIGH: "Price Too High",
  COMPETITOR_RELATION: "Competitor Relationship",
  PRODUCT_MISMATCH: "Product Mismatch",
  TIMELINE_ISSUE: "Timeline Issue",
  NO_BUDGET: "No Budget",
  NO_RESPONSE: "No Response",
  COMPETITOR_PROMO: "Competitor Promotion",
  DISTRIBUTOR_CONFLICT: "Distributor Conflict",
  CAPACITY_SUPPLY: "Capacity / Supply Issue",
  INTERNAL_FOLLOWUP: "Internal Follow-up Failure",
  OTHER: "Other",
};

export const WIN_REASON_LABELS: Record<CRMWinReason, string> = {
  BETTER_PRICE: "Better Price",
  BETTER_PRODUCT_FIT: "Better Product Fit",
  FASTER_RESPONSE: "Faster Response",
  STRONGER_RELATION: "Stronger Relationship",
  AVAILABILITY: "Product Availability",
  TECHNICAL_SUPPORT: "Technical Support",
  PROMOTION_SUPPORT: "Promotion Support",
  OTHER: "Other",
};

export const AI_AGENT_LABELS: Record<CRMAIAgentType, string> = {
  LEAD_PRIORITIZATION: "Lead Prioritization Assistant",
  PIPELINE_RISK: "Pipeline Risk Monitor",
  WIN_LOSS_INSIGHT: "Win/Loss Insight Assistant",
};

export function fmtCurrency(val?: number | null, currency = "KES"): string {
  if (val == null) return "—";
  return `${currency} ${val.toLocaleString("en-KE", { minimumFractionDigits: 0, maximumFractionDigits: 0 })}`;
}

// ── API Client ─────────────────────────────────────────────────────────────────

export const crmApi = {
  // Stages
  getStages: (activeOnly = false) => r(ax.get<CRMStage[]>(`/crm/stages`, { params: { active_only: activeOnly } })),
  createStage: (data: Partial<CRMStage>) => r(ax.post<CRMStage>(`/crm/stages`, data)),
  updateStage: (id: string, data: Partial<CRMStage>) => r(ax.patch<CRMStage>(`/crm/stages/${id}`, data)),
  seedDefaultStages: () => r(ax.post<CRMStage[]>(`/crm/stages/seed-defaults`)),

  // Records
  getLeads: (params?: Record<string, unknown>) => r(ax.get<CRMRecord[]>(`/crm/leads`, { params })),
  createLead: (data: Partial<CRMRecord>) => r(ax.post<CRMRecord>(`/crm/leads`, data)),
  getOpportunities: (params?: Record<string, unknown>) => r(ax.get<CRMRecord[]>(`/crm/opportunities`, { params })),
  createOpportunity: (data: Partial<CRMRecord>) => r(ax.post<CRMRecord>(`/crm/opportunities`, data)),
  getAllRecords: (params?: Record<string, unknown>) => r(ax.get<CRMRecord[]>(`/crm/records`, { params })),
  getRecord: (id: string) => r(ax.get<CRMRecord>(`/crm/records/${id}`)),
  updateRecord: (id: string, data: Partial<CRMRecord>) => r(ax.patch<CRMRecord>(`/crm/records/${id}`, data)),

  // Lifecycle
  qualify: (id: string, data: Record<string, unknown>) => r(ax.post<CRMRecord>(`/crm/records/${id}/qualify`, data)),
  convert: (id: string, data: Record<string, unknown>) => r(ax.post<CRMRecord>(`/crm/records/${id}/convert-to-opportunity`, data)),
  closeWon: (id: string, data: Record<string, unknown>) => r(ax.post<CRMRecord>(`/crm/records/${id}/close-won`, data)),
  closeLost: (id: string, data: Record<string, unknown>) => r(ax.post<CRMRecord>(`/crm/records/${id}/close-lost`, data)),
  onHold: (id: string, notes?: string) => r(ax.post<CRMRecord>(`/crm/records/${id}/on-hold`, null, { params: { notes } })),
  reopen: (id: string) => r(ax.post<CRMRecord>(`/crm/records/${id}/reopen`)),
  updateScore: (id: string) => r(ax.post<CRMRecord>(`/crm/records/${id}/update-score`)),

  // Interest lines
  addInterestLine: (recordId: string, data: Partial<CRMInterestLine>) =>
    r(ax.post<CRMInterestLine>(`/crm/records/${recordId}/interest-lines`, data)),
  deleteInterestLine: (lineId: string) => ax.delete(`/crm/interest-lines/${lineId}`),

  // Activities
  getActivities: (params?: Record<string, unknown>) => r(ax.get<CRMActivity[]>(`/crm/activities`, { params })),
  createActivity: (data: Partial<CRMActivity>) => r(ax.post<CRMActivity>(`/crm/activities`, data)),
  completeActivity: (id: string, data: Record<string, unknown>) =>
    r(ax.post<CRMActivity>(`/crm/activities/${id}/complete`, data)),

  // Competitors
  addCompetitor: (data: Partial<CRMCompetitor>) => r(ax.post<CRMCompetitor>(`/crm/competitors`, data)),

  // Deduplication
  checkDuplicates: (companyName: string, email?: string, phone?: string) =>
    r(ax.get<CRMRecord[]>(`/crm/check-duplicates`, { params: { company_name: companyName, email, phone } })),

  // Dashboard & Reports
  getDashboard: () => r(ax.get<CRMDashboard>(`/crm/dashboard`)),
  getForecast: (monthsAhead = 6) => r(ax.get<Record<string, unknown>[]>(`/crm/forecast`, { params: { months_ahead: monthsAhead } })),
  getPipelineReport: () => r(ax.get<Record<string, unknown>[]>(`/crm/reports/pipeline`)),
  getWinLossReport: () => r(ax.get<Record<string, unknown>>(`/crm/reports/win-loss`)),

  // AI
  runLeadPrioritization: () => r(ax.post<{ generated: number; message: string }>(`/crm/ai/run-lead-prioritization`)),
  runPipelineRisk: () => r(ax.post<{ generated: number; message: string }>(`/crm/ai/run-pipeline-risk`)),
  runWinLossInsight: () => r(ax.post<{ generated: number; message: string }>(`/crm/ai/run-win-loss-insight`)),
  getAIRecs: (params?: Record<string, unknown>) => r(ax.get<CRMAIRec[]>(`/crm/ai/recommendations`, { params })),
  ackAIRec: (id: string, data: Record<string, unknown>) => r(ax.patch<CRMAIRec>(`/crm/ai/recommendations/${id}`, data)),
};

File: backend/app/api/v1/endpoints/crm_pipeline.py
from __future__ import annotations
from typing import List, Optional
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.db.session import get_db
from app.schemas.crm import (
    CRMStageCreate, CRMStageRead, CRMStageUpdate,
    CRMRecordCreate, CRMRecordRead, CRMRecordUpdate,
    CRMInterestLineCreate, CRMInterestLineRead,
    CRMActivityCreate, CRMActivityRead, CRMActivityComplete,
    CRMCompetitorCreate, CRMCompetitorRead,
    CRMQualifyRequest, CRMConvertRequest,
    CRMCloseWonRequest, CRMCloseLostRequest,
    CRMWinLossRead,
    CRMAIRecRead, CRMAIRecAck,
)
import app.services.crm_pipeline_service as svc

router = APIRouter()


# ── Stages ────────────────────────────────────────────────────────────────────

@router.get("/stages", response_model=List[CRMStageRead])
async def list_stages(active_only: bool = False, db: AsyncSession = Depends(get_db)):
    return await svc.get_stages(db, active_only=active_only)


@router.post("/stages", response_model=CRMStageRead, status_code=201)
async def create_stage(data: CRMStageCreate, db: AsyncSession = Depends(get_db)):
    return await svc.create_stage(db, data)


@router.patch("/stages/{stage_id}", response_model=CRMStageRead)
async def update_stage(stage_id: str, data: CRMStageUpdate, db: AsyncSession = Depends(get_db)):
    result = await svc.update_stage(db, stage_id, data)
    if not result:
        raise HTTPException(404, "Stage not found")
    return result


@router.post("/stages/seed-defaults", response_model=List[CRMStageRead])
async def seed_default_stages(db: AsyncSession = Depends(get_db)):
    return await svc.seed_default_stages(db)


# ── Records ───────────────────────────────────────────────────────────────────

@router.get("/leads", response_model=List[CRMRecordRead])
async def list_leads(
    status: Optional[str] = None,
    assigned_rep_id: Optional[str] = None,
    stage_id: Optional[str] = None,
    temperature: Optional[str] = None,
    skip: int = 0,
    limit: int = 100,
    db: AsyncSession = Depends(get_db),
):
    return await svc.get_records(db, record_type="LEAD", status=status,
                                  assigned_rep_id=assigned_rep_id, stage_id=stage_id,
                                  temperature=temperature, skip=skip, limit=limit)


@router.post("/leads", response_model=CRMRecordRead, status_code=201)
async def create_lead(data: CRMRecordCreate, db: AsyncSession = Depends(get_db)):
    from app.models.crm import CRMRecordType
    data.record_type = CRMRecordType.LEAD
    return await svc.create_record(db, data)


@router.get("/opportunities", response_model=List[CRMRecordRead])
async def list_opportunities(
    status: Optional[str] = None,
    assigned_rep_id: Optional[str] = None,
    stage_id: Optional[str] = None,
    temperature: Optional[str] = None,
    skip: int = 0,
    limit: int = 100,
    db: AsyncSession = Depends(get_db),
):
    return await svc.get_records(db, record_type="OPPORTUNITY", status=status,
                                  assigned_rep_id=assigned_rep_id, stage_id=stage_id,
                                  temperature=temperature, skip=skip, limit=limit)


@router.post("/opportunities", response_model=CRMRecordRead, status_code=201)
async def create_opportunity(data: CRMRecordCreate, db: AsyncSession = Depends(get_db)):
    from app.models.crm import CRMRecordType
    data.record_type = CRMRecordType.OPPORTUNITY
    return await svc.create_record(db, data)


@router.get("/records", response_model=List[CRMRecordRead])
async def list_all_records(
    record_type: Optional[str] = None,
    status: Optional[str] = None,
    assigned_rep_id: Optional[str] = None,
    stage_id: Optional[str] = None,
    temperature: Optional[str] = None,
    skip: int = 0,
    limit: int = 100,
    db: AsyncSession = Depends(get_db),
):
    return await svc.get_records(db, record_type=record_type, status=status,
                                  assigned_rep_id=assigned_rep_id, stage_id=stage_id,
                                  temperature=temperature, skip=skip, limit=limit)


@router.post("/records", response_model=CRMRecordRead, status_code=201)
async def create_record(data: CRMRecordCreate, db: AsyncSession = Depends(get_db)):
    return await svc.create_record(db, data)


@router.get("/records/{record_id}", response_model=CRMRecordRead)
async def get_record(record_id: str, db: AsyncSession = Depends(get_db)):
    rec = await svc.get_record(db, record_id)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


@router.patch("/records/{record_id}", response_model=CRMRecordRead)
async def update_record(record_id: str, data: CRMRecordUpdate, db: AsyncSession = Depends(get_db)):
    rec = await svc.update_record(db, record_id, data)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


@router.post("/records/{record_id}/qualify", response_model=CRMRecordRead)
async def qualify_record(record_id: str, data: CRMQualifyRequest, db: AsyncSession = Depends(get_db)):
    rec = await svc.qualify_record(db, record_id, data)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


@router.post("/records/{record_id}/convert-to-opportunity", response_model=CRMRecordRead)
async def convert_to_opportunity(record_id: str, data: CRMConvertRequest, db: AsyncSession = Depends(get_db)):
    rec = await svc.convert_to_opportunity(db, record_id, data)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


@router.post("/records/{record_id}/close-won", response_model=CRMRecordRead)
async def close_won(record_id: str, data: CRMCloseWonRequest, db: AsyncSession = Depends(get_db)):
    rec = await svc.close_won(db, record_id, data)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


@router.post("/records/{record_id}/close-lost", response_model=CRMRecordRead)
async def close_lost(record_id: str, data: CRMCloseLostRequest, db: AsyncSession = Depends(get_db)):
    rec = await svc.close_lost(db, record_id, data)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


@router.post("/records/{record_id}/on-hold", response_model=CRMRecordRead)
async def put_on_hold(record_id: str, notes: Optional[str] = None, db: AsyncSession = Depends(get_db)):
    rec = await svc.put_on_hold(db, record_id, notes)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


@router.post("/records/{record_id}/reopen", response_model=CRMRecordRead)
async def reopen_record(record_id: str, db: AsyncSession = Depends(get_db)):
    rec = await svc.reopen_record(db, record_id)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


@router.post("/records/{record_id}/update-score", response_model=CRMRecordRead)
async def update_lead_score(record_id: str, db: AsyncSession = Depends(get_db)):
    rec = await svc.update_lead_score(db, record_id)
    if not rec:
        raise HTTPException(404, "Record not found")
    return rec


# ── Interest Lines ────────────────────────────────────────────────────────────

@router.post("/records/{record_id}/interest-lines", response_model=CRMInterestLineRead, status_code=201)
async def add_interest_line(record_id: str, data: CRMInterestLineCreate, db: AsyncSession = Depends(get_db)):
    return await svc.add_interest_line(db, data)


@router.delete("/interest-lines/{line_id}", status_code=204)
async def delete_interest_line(line_id: str, db: AsyncSession = Depends(get_db)):
    if not await svc.delete_interest_line(db, line_id):
        raise HTTPException(404, "Interest line not found")


# ── Activities ────────────────────────────────────────────────────────────────

@router.get("/activities", response_model=List[CRMActivityRead])
async def list_activities(
    record_id: Optional[str] = None,
    assigned_to: Optional[str] = None,
    result_status: Optional[str] = None,
    overdue_only: bool = False,
    skip: int = 0,
    limit: int = 200,
    db: AsyncSession = Depends(get_db),
):
    return await svc.get_activities(db, record_id=record_id, assigned_to=assigned_to,
                                     result_status=result_status, overdue_only=overdue_only,
                                     skip=skip, limit=limit)


@router.post("/activities", response_model=CRMActivityRead, status_code=201)
async def create_activity(data: CRMActivityCreate, db: AsyncSession = Depends(get_db)):
    return await svc.create_activity(db, data)


@router.post("/activities/{activity_id}/complete", response_model=CRMActivityRead)
async def complete_activity(activity_id: str, data: CRMActivityComplete, db: AsyncSession = Depends(get_db)):
    act = await svc.complete_activity(db, activity_id, data)
    if not act:
        raise HTTPException(404, "Activity not found")
    return act


# ── Competitors ───────────────────────────────────────────────────────────────

@router.post("/competitors", response_model=CRMCompetitorRead, status_code=201)
async def add_competitor(data: CRMCompetitorCreate, db: AsyncSession = Depends(get_db)):
    return await svc.add_competitor(db, data)


# ── Deduplication ─────────────────────────────────────────────────────────────

@router.get("/check-duplicates", response_model=List[CRMRecordRead])
async def check_duplicates(
    company_name: str,
    email: Optional[str] = None,
    phone: Optional[str] = None,
    db: AsyncSession = Depends(get_db),
):
    return await svc.check_duplicates(db, company_name, email, phone)


# ── Dashboard & Reports ───────────────────────────────────────────────────────

@router.get("/dashboard")
async def get_dashboard(db: AsyncSession = Depends(get_db)):
    return await svc.get_dashboard(db)


@router.get("/forecast")
async def get_forecast(months_ahead: int = 6, db: AsyncSession = Depends(get_db)):
    return await svc.get_forecast(db, months_ahead=months_ahead)


@router.get("/reports/pipeline")
async def pipeline_report(db: AsyncSession = Depends(get_db)):
    return await svc.pipeline_report(db)


@router.get("/reports/win-loss")
async def win_loss_report(db: AsyncSession = Depends(get_db)):
    return await svc.win_loss_report(db)


# ── AI Agents ─────────────────────────────────────────────────────────────────

@router.post("/ai/run-lead-prioritization")
async def run_lead_prioritization(db: AsyncSession = Depends(get_db)):
    recs = await svc.run_lead_prioritization(db)
    return {"generated": len(recs), "message": f"Lead Prioritization agent generated {len(recs)} recommendations"}


@router.post("/ai/run-pipeline-risk")
async def run_pipeline_risk(db: AsyncSession = Depends(get_db)):
    recs = await svc.run_pipeline_risk(db)
    return {"generated": len(recs), "message": f"Pipeline Risk agent generated {len(recs)} recommendations"}


@router.post("/ai/run-win-loss-insight")
async def run_win_loss_insight(db: AsyncSession = Depends(get_db)):
    recs = await svc.run_win_loss_insight(db)
    return {"generated": len(recs), "message": f"Win/Loss Insight agent generated {len(recs)} recommendations"}


@router.get("/ai/recommendations", response_model=List[CRMAIRecRead])
async def get_ai_recommendations(
    agent_type: Optional[str] = None,
    status: Optional[str] = None,
    db: AsyncSession = Depends(get_db),
):
    return await svc.get_ai_recommendations(db, agent_type=agent_type, status=status)


@router.patch("/ai/recommendations/{rec_id}", response_model=CRMAIRecRead)
async def ack_recommendation(rec_id: str, data: CRMAIRecAck, db: AsyncSession = Depends(get_db)):
    rec = await svc.ack_ai_recommendation(db, rec_id, data)
    if not rec:
        raise HTTPException(404, "Recommendation not found")
    return rec
